File: src/main/java/com/example/demo/service/PaymentService.java
// ПУТЬ: src/main/java/com/example/demo/service/PaymentService.java

package com.example.demo.service;

import com.example.demo.dto.request.PaymentCreateRequestDTO;
import com.example.demo.dto.response.PaymentResponseDTO;
import java.math.BigDecimal;
import java.util.List;

public interface PaymentService {
    PaymentResponseDTO createPayment(PaymentCreateRequestDTO request, Long userId);
    PaymentResponseDTO getPaymentById(Long id);
    PaymentResponseDTO getPaymentByReservationId(Long reservationId);
    List<PaymentResponseDTO> getUserPayments(Long userId);
    void completePayment(Long id);
    void refundPayment(Long id, String reason);
    BigDecimal calculateRefundAmount(Long reservationId);
}

File: frontend/src/services/paymentService.js
import api from './api';

export const paymentService = {
  createPayment: (data) => api.post('/payments', data),
  getPaymentById: (id) => api.get(`/payments/${id}`),
  getPaymentByReservation: (reservationId) => api.get(`/payments/reservation/${reservationId}`),
  getUserPayments: (userId) => api.get(`/payments/user/${userId}`),
  completePayment: (id) => api.put(`/payments/${id}/complete`),
  refundPayment: (id, reason) => api.put(`/payments/${id}/refund`, null, { params: { reason } }),
  calculateRefundAmount: (reservationId) => api.get(`/payments/reservation/${reservationId}/refund-amount`),
};
